src/puppetlabs/trapperkeeper/services/status/status_service.clj

```diff
@@ -33,6 +33,11 @@
       (add-ring-handler this handler))
     context)
 
+  (stop [this context]
+    (status-core/dissoc-status-context! (:status-fns context)
+                                        status-core/status-service-name)
+    context)
```

Comment:
re: 2., that kind of seems like a bug in the file sync test then (or wherever you got that from).  I suspect that if you change that to 'restart' your test will stop failing, and that would unblock you on that stream of work.

That's not to say that this PR isn't still a good idea, but it would no longer be a blocker, AFAIK.